deploy-0chain/action.yml

```diff
@@ -176,61 +170,126 @@ runs:
+                    echo "failed to run $@"
+                    cat build.log && echo "::error title=Failed to stake the blobbers";
+                    echo "exiting blobber stake process..."
+                    exit 1
+                fi
+            }
+            done
+        }
+        retry ./zwallet register --configDir ./config --config ./zbox_config.yaml --wallet ../ignore > build.log 2>&1
+        for i in {1..6}
+        do
+            retry ./zwallet faucet --methodName "pour" --input "{}" --tokens 2 --configDir ./config --config ./zbox_config.yaml --wallet ../ignore > build.log 2>&1
+        done
 
-          if ( kubectl -n ${{ env.NAMESPACE }} get pods --kubeconfig ./kube/${{ env.NAMESPACE }}-config | grep blobber-stake | grep -i Error ); then
-            echo "::$ANNOTATION_LEVEL title=0Chain network deployment failed on $RETRY_ATTEMPT attempt::blobber stake failed on $RETRY_ATTEMPT attempt"
-            exit 1
-          fi
+        retry ./zwallet getblobbers --silent --configDir ./config --config ./zbox_config.yaml --wallet ../ignore | grep -o "[a-z0-9]\{64\}" > build.log
+        readarray -t blobber_ids < build.log
+        for id in "${blobber_ids[@]}"
+        do
+            retry ./zbox sp-lock --blobber_id $id --tokens 1 --configDir ./config --config ./zbox_config.yaml --wallet ../ignore > build.log 2>&1
+        done
+        ' > ./stake_blobbers.sh;
+        chmod 777 ./stake_blobbers.sh;
 
-          sleep 10
-          echo "Still waiting for 0Chain deployment to finish..." && ((counter=counter+1));
+        ITER=0
+        MAX_RETRIES=3
+        until [[ $ITER -eq $MAX_RETRIES ]]
+        do
+            ./stake_blobbers.sh && break
+            ITER=$((ITER+1))
+            echo "Staking blobbers failed on try $ITER. Retrying in 5 seconds..."
+            sleep 5
         done
 
-        if [ $counter -eq 30 ]; then
-          echo "::$ANNOTATION_LEVEL title=0Chain network deployment failed on $RETRY_ATTEMPT attempt::Chain setup did not complete within the time limit on $RETRY_ATTEMPT attempt"
-          exit 1
+        if [[ $ITER -eq $MAX_RETRIES ]]
+          then
+            echo "Staking blobbers failed after $MAX_RETRIES tries. Exiting run..."
+            exit 1
+          else
+            echo "Blobbers staked succefully on try $((ITER+1)). Moving on to tests..."
         fi
 
-        echo "::notice title=0Chain network deployment was successful on $RETRY_ATTEMPT attempt::0Chain was successfully deployed to [${{ env.NETWORK_URL }}]"
-        ' > ./wait.sh;
-        chmod 777 ./wait.sh;
-        ./wait.sh "first" || {
-          echo '#!/bin/bash
-          RED="\033[0;31m"
-          NC="\033[0m"
-          echo
-          echo -e "${RED}======================================================${NC}"
-          echo -e "${RED}RETRYING DEPLOYMENT ($(echo $1 | awk '"'"'{ print toupper($0) }'"'"') ATTEMPT):${NC}"
```

Comment:
reminder for myself: network deploy retries need re-added

Reply:
Done: Here's what a purposefully failed run looks like now: https://github.com/0chain/system_test/runs/5727416719?check_suite_focus=true